File: Programmers/점찍기.py
import math
"""
좌표평면이란게 중요함.
좌표평면에서 거리를 통해 나타낼 수 있는 도형이 원이고, 원은
x**2 + y**2 = d**2라는것을 알면
y의 방정식으로 바꿔서
y = math.sqrt(d**2-x**2)로 해서
0~d+1까지의 y에 대해 몇개의 좌표가 찍히는지 알 수 있다.

"""
def solution(k, d):
    answer=0
    for i in range(0,d+1,k):
        y = math.sqrt(d**2-i**2)
        answer+=math.floor(y/k)+1
    return answer

File: Programmers/점찍기.java
/*
"원점과 거리"가 d이하인 x,y좌표의 경우의 수를 return하는 문제.
원점과의 거리가 일정한 집합 = 원.
x**2 + y**2 <= d**2
-> 0~d까지 범위를 x로 대입하고, y에 대해 방정식을 세워서 y의 갯수를 answer에 더해주는 방식으로 푼다.
해당 y의 갯수 또한 k의 간격으로만 존재하므로 Math.floor(y/k)+1씩 더해주면 된다.
*/
class Solution {
    public long solution(int k, int d) {
        long answer = 0;
        
        for (int i=0;i<=d;i+=k){
            int y = (int)Math.sqrt(Math.pow(d,2)-Math.pow(i,2));
            answer += Math.floor(y/k)+1;
        }
        return answer;
    }
}
